### PR TITLE
fail hardening

### PR DESCRIPTION
**Pull request by marsch**
added missing test and debugs issue with that: https://github.com/easybib/issues/issues/3303

### CODE REVIEW

**Comment by marsch** [comment]
@sashless wanna review? i just added one test and you were right, it wasn't rejected correctly - now this should work fine

File: src/easybib-api-client_spec.js [hunk 1]
```diff
@@ -215,5 +217,39 @@ describe('EasyBib Api Client', function() {
       $rootScope.$digest();
       $httpBackend.flush(2, false);
     });
+
+    it('should reject if retries also fail', function(done) {
+
+      localStorage.removeItem('easybib-api-access-data');
+      easybibApiClient.get('http://noopurl.notld/citations')
+      .then(function() {
+        assert.ok(0, 'request was not rejected');
```
Comment (sashless) [inline src/easybib-api-client_spec.js]: can we please add a comment to explain why you have added that ?

File: src/easybib-api-client.js [hunk 2]
```diff
@@ -97,6 +98,8 @@ module.exports = function($q, $http, store, $timeout, easyBibApiAccessUrl) {
         // jscs:enable
         self.request(req).then(function(response) {
           deferred.resolve(response);
+        }, function(error) {
```
Comment (sashless) [inline src/easybib-api-client.js]: i think this can be replaced with

```
 return  self.request(req);
```

Comment (sashless) [inline src/easybib-api-client.js]: and then in line https://github.com/easybiblabs/easybib-api-client-angular/pull/4/files#diff-5dc66d40dc610fe7157d0c36f8184bdeR94

do 

``` 
return    utils.getAccessToken();
```

then you can remove the extra deffered

**Pull request merged**